### PR TITLE
[adobe_analytics] Added support for ACPIdentity's appendVisitorInfo(for: URL)

### PR DESCRIPTION
**Pull request by jeremie-movify**
Implemented for both Android & iOS + added matching tests

### CODE REVIEW

**Review by timrijckaert — APPROVED** [review]

File: packages/adobe_analytics/ios/Classes/SwiftAdobeAnalyticsPlugin.swift [hunk 1]
```diff
@@ -66,21 +71,27 @@ extension SwiftAdobeAnalyticsPlugin {
...
-
+  
   public func trackState(_ state: String, data: [String : String]?) {
     ACPCore.trackState(state, data: data)
   }
-
+  
+  public func appendVisitorInfo(to url: URL, completion: @escaping (String) -> Void) {
+    ACPIdentity.append(to: url) { updatedURL in
+      completion(updatedURL!.absoluteString)
```
Comment (Kymer) [inline packages/adobe_analytics/ios/Classes/SwiftAdobeAnalyticsPlugin.swift]: Given Adobe's blackbox I wouldn't rely on `updatedURL` never being nil. (maybe if the SDK isn't properly initialised the returned URL's are nil, for example). So why not do `updatedURL?.absoluteString ?? url.absoluteString`?

File: packages/adobe_analytics/lib/adobe_analytics.dart [hunk 2]
```diff
@@ -21,6 +21,9 @@ class AdobeAnalytics {
   Future<bool> trackState(String action, {Map<String, String> data}) =>
       _channel.invokeMethod('track', {'type': 'state', 'key': action, 'data': data});
 
-  /// Retrieves the Experience Cloud ID.
+  /// Retrieve the Experience Cloud ID.
   Future<String> getExperienceCloudId() => _channel.invokeMethod('getExperienceCloudId');
+
+  /// Append visitor information to the given URL.
+  Future<String> appendVisitorInfo(String url) => _channel.invokeMethod('appendVisitorInfo', {'url': url});
```
Comment (Kymer) [inline packages/adobe_analytics/lib/adobe_analytics.dart]: How does the Android side handle an empty url String? The Swift implementation will throw a decoding error, but does the logging make sense for someone using the plugin? As in: is it a clear, actionable error?

Perhaps a _not empty_ check on Dart side could help if the native implementations don't handle it gracefully? ¯\\\_(ツ)_/¯

File: packages/adobe_analytics/ios/Classes/SwiftAdobeAnalyticsPlugin.swift (hunk 1 above)
Comment (jeremie-movify) [reply packages/adobe_analytics/ios/Classes/SwiftAdobeAnalyticsPlugin.swift]: Good idea ; I'll update the PR 👍

File: packages/adobe_analytics/lib/adobe_analytics.dart (hunk 2 above)
Comment (jeremie-movify) [reply packages/adobe_analytics/lib/adobe_analytics.dart]: Indeed, that's a good idea ; I'll `assert` the `url` parameter

**Review by timrijckaert — APPROVED** [review]

**Review by timrijckaert — APPROVED** [review]

**Review by Kymer — APPROVED** [review]

**Pull request merged**